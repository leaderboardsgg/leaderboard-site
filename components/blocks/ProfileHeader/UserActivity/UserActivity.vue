<template>
  <div class="user-activity" v-bind="$attrs">
    <p><span>Runs: </span>{{ runCount }}</p>
    <p><span>Games: </span>{{ gameCount }}</p>
  </div>
</template>

<script lang="ts">
import { defineComponent } from '@nuxtjs/composition-api';

export default defineComponent({
  name: 'UserActivity',
  props: {
    gameCount: {
      default: () => 6,
      type: Number,
    },
    runCount: {
      default: () => 42,
      type: Number,
    },
  },
});
</script>

<style scoped>
.user-activity {
  & span {
    @apply font-semibold;
  }
}
</style>
