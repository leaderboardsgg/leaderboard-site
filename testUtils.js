import Vue from 'vue';
import { RouterLinkStub } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (Component, options) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
