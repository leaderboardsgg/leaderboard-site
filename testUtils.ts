import Vue, { ComponentOptions } from 'vue';
import { RouterLinkStub, VueClass } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render, RenderOptions } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component<
  { tag: string; to: string },
  unknown,
  unknown
>('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data?.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (
  Component: VueClass<Vue> | ComponentOptions<Vue>,
  options?: RenderOptions<Vue>,
) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
