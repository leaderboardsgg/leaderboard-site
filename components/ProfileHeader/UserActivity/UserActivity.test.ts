import { stubbedRender } from '@/testUtils';
import UserActivity from './UserActivity.vue';

describe('<UserActivity />', () => {
  it('should render without crashing', () => {
    const { unmount } = stubbedRender(UserActivity);

    unmount();
  });

  it('renders correctly', () => {
    const { container } = stubbedRender(UserActivity);

    expect(container.firstChild).toMatchSnapshot();
  });
});
