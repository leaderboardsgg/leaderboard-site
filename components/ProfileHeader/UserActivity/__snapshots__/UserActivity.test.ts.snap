// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserActivity /> renders correctly 1`] = `
<div
  class="profile-info--user-activity"
>
  <p>
    <span
      class="font-semibold"
    >
      Runs: 
    </span>
    42
  </p>
   
  <p>
    <span
      class="font-semibold"
    >
      Games: 
    </span>
    6
  </p>
</div>
`;
